scheduler/src/cook/mesos/task.clj

```diff
@@ -111,14 +112,14 @@
 (defn take-all-scalar-resources-for-task
   [resources task]
   (reduce
-   (fn [{:keys [remaining-resources mesos-messages]} [resource-keyword amount]]
-     (let [resource-name (name resource-keyword)
-           adjustment (take-resources remaining-resources resource-name amount)]
-       {:remaining-resources (assoc remaining-resources resource-name (:remaining-resources adjustment))
-        :mesos-messages (into mesos-messages (:mesos-messages adjustment))}))
-   {:remaining-resources resources
-    :mesos-messages []}
-   (-> task :resources vec)))
+    (fn [{:keys [remaining-resources mesos-messages]} [resource-keyword amount]]
+      (let [resource-name (name resource-keyword)
+            adjustment (take-resources remaining-resources resource-name amount)]
+        {:remaining-resources (assoc remaining-resources resource-name (:remaining-resources adjustment))
+         :mesos-messages (into mesos-messages (:mesos-messages adjustment))}))
+    {:remaining-resources resources
+     :mesos-messages []}
+    (.getScalarRequests ^com.netflix.fenzo.TaskRequest (:task-request task))))
```

Comment:
I think that while many changes were made moving to new data formats, the people making the changes didn't think about how far reaching they might need to be in their adaptations. I think that the move to Mesomatic & understanding resources as collections of tagged resources (with multiple CPU resources, mem resources on a single machine, etc) should have also change the existing aggregation-based code where needed. Here's a place it doesn't seem to quite line up.